libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -780,10 +814,16 @@ bool ContractCompiler::visit(Return const& _return)
 		for (auto const& retVariable: boost::adaptors::reverse(returnParameters))
 			CompilerUtils(m_context).moveToStackVariable(*retVariable);
 	}
-	for (unsigned i = 0; i < m_stackCleanupForReturn; ++i)
-		m_context << Instruction::POP;
-	m_context.appendJumpTo(m_returnTags.back());
-	m_context.adjustStackOffset(m_stackCleanupForReturn);
+
+	unsigned sizeOnStack = 0;
+	for (auto _varDecl : m_scopedVariables)
+		for (auto _decl : _varDecl.second)
+			sizeOnStack += _decl->type()->sizeOnStack();
+	m_returnTagsPops.push_back(std::make_pair<unsigned, eth::AssemblyItem>(
```

Comment:
Why don't you just remove the local variables in the same way it was done before?

Reply:
To use the optimized shared structure thingy.